app/Platform/Concerns/BuildsGameListQueries.php

```diff
@@ -323,18 +323,25 @@ private function applyReleasedAtSorting(Builder $query, string $sortDirection =
         // and SQLite. This is preferable to altering the query specifically for
         // SQLite, because if we do so then we can't actually trust any test results.
         $query
-            ->selectRaw(
-                "GameData.*,
+            ->selectRaw(<<<SQL
+                GameData.*,
                 CASE
                     WHEN GameData.released_at_granularity = 'year' THEN
                         DATE(CONCAT(SUBSTR(GameData.released_at, 1, 4), '-01-01'))
                     WHEN GameData.released_at_granularity = 'month' THEN
                         DATE(CONCAT(SUBSTR(GameData.released_at, 1, 7), '-01'))
                     ELSE
                         COALESCE(GameData.released_at, '9999-12-31')
-                END AS normalized_released_at"
-            )
-            ->orderBy('normalized_released_at', $sortDirection);
+                END AS normalized_released_at,
+                CASE GameData.released_at_granularity
+                    WHEN 'year' THEN 1
+                    WHEN 'month' THEN 2
+                    WHEN 'day' THEN 3
+                    ELSE 4
+                END AS granularity_order
```

Comment:
I was momentarily confused by this putting day-granular results in the middle of month-granular results:
![image](https://github.com/user-attachments/assets/8b70cf7e-c0c7-41fc-9246-172c4f7fce1e)
Notice the day granular items in between the January and September monthly granularities?

Digging into it, it's caused by the primary sort on `normalized_released_at`:
![image](https://github.com/user-attachments/assets/d59ad606-dab4-4cbc-94c0-34452b5720f4)

Which makes sense - the granularity sort is just being used as a tie-breaker.

However, are there concerns about the day-granular values having timestamp information?

Reply:
No concerns from me. In MariaDB, if the time isn't set, it's implied as 00:00:00. I'm not sure where the nuance is on how it's getting set - perhaps a difference from the sync script and Filament modifications.

Reply:
It's from the query. When you construct the dates for the less granular options, you don't include the timestamp. The most granular option is read directly from the DB without any alterations.

Reply:
Ah, I see what's happening now.

I do have a change locally which causes the query to always return timestamps:
![Screenshot 2025-01-16 at 7 24 40 PM](https://github.com/user-attachments/assets/1cfbc0f7-a290-4bcf-9d70-8687da178372)

but, if the query is currently causing results to be displayed in the expected fashion, I'll hang on to this in a stash unless we'd like to have it in `master`.